src/app/backend/resource/secret/secretdetail.go

```diff
@@ -51,11 +52,24 @@ func GetSecretDetail(client *client.Client, namespace, name string) (*SecretDeta
 	return getSecretDetail(rawSecret), nil
 }
 
+func hideSecretData(rawSecret *api.Secret) map[string][]byte {
+	// For opaque and basicAuth secret types the credentials like username, password and etc. should be hidden.
+	if rawSecret.Type == api.SecretTypeOpaque || rawSecret.Type == api.SecretTypeBasicAuth {
+		data := make(map[string][]byte)
+		for key, value := range rawSecret.Data {
+			data[key] = []byte(fmt.Sprintf("%d bytes", len(value)))
```

Comment:
I agree, the best option seems to be adding switch like `(show value)` and hiding value by default.